src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -72,6 +72,8 @@ static bool haveMutationsOfDynamicColumns(const MergeTreeData::DataPartPtr & dat
             if (column && column->type->hasDynamicSubcolumns())
                 return true;
         }
+        if (command.type == MutationCommand::Type::MATERIALIZE_COLUMNS)
+            return true;
```

Comment:
TBH I'm not sure about it. If we have Dynamic columns in table, but they don't have any special DEFAULT, than MATERIALIZE_COLUMNS shouldn't affect them, right?

Reply:
yeah, right. we should not affect those. 
This change seems redundant. initially I want to make the caller (`splitAndModifyMutationCommands`) to be always true for `MATERIALIZE_COLUMNS` . but seems like this is not needed, I debugged as the `isWidePart()` is not true in splitAndModifyMutationCommands (can be compact). let me remove this change.